src/main/capi/data_chunk-c.cpp

```diff
@@ -128,6 +128,22 @@ idx_t duckdb_list_vector_get_size(duckdb_vector vector) {
 	return duckdb::ListVector::GetListSize(*v);
 }
 
+void duckdb_list_vector_set_size(duckdb_vector vector, idx_t size) {
+	if (!vector) {
+		return;
```

Comment:
Since this doesn't have a desired return value, could we perhaps make it a `duckdb_state` so we can indicate that the input was faulty instead of silently returning?

Same goes for the `duckdb_list_vector_reserve`

Reply:
sure thing. will do.